src/TwigComponent/src/Twig/TwigPreLexer.php

```diff
@@ -35,10 +35,27 @@ public function __construct(int $startingLine = 1)
 
     public function preLexComponents(string $input): string
     {
-        if (!str_contains($input, '<twig:')) {
+        // tag may be:
+        // - classical: <twig:componentName>
+        // - jsx: <ComponentName> (with a capital letter)
+
+        $isClassical = str_contains($input, '<twig:');
+        $isJsx = preg_match_all('/<([A-Z][a-zA-Z0-9_:-]+)([^>]*)>/', $input, $matches, \PREG_SET_ORDER);
```

Comment:
I don't think if we can really qualify it as JSX, the properties and spread syntaxes are different and here only the naming case is the same.
What about "$isShort" and "$isLong"?

Reply:
can it be both? Then `has` is better as `is`: https://3v4l.org/Wln8L

Reply:
Thanks! To me, it doesn't seem relevant to use a has naming here. The goal is to detect which syntax is being used, not whether both are present.

If we ever introduce support for mixing syntaxes or handling them differently in the same input, then switching to a `hasShortSyntax` / `hasLongSyntax` style might make sense. But for now, a simple `$isShort` better reflects the current logic and avoids overcomplicating things prematurely.